### PR TITLE
Applying file based renames to primary outputs

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes https://github.com/dotnet/templating/issues/2453 fixes https://github.com/dotnet/templating/issues/2682 by applying symbol based renames to primary outputs defined in configuration.

The solution to be discussed: it assumes that primary output should contain path to final file before applying symbol based renames (different from current definition of primaryOutputs in [Wiki ](https://github.com/dotnet/templating/wiki/Reference-for-template.json#primary-output-definition)

Integration tests are covering the fix.
Yet all not tests can be enabled due to the bug that `ICreationEffects.FileChanges` do not contain valid source and target paths if source modifiers are applied.

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs
Comment (donJoseLuis) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs:49]: minor:
during code refactoring efforts, I'd like to adopt a 1 param per line style when method signatures have more then 3-5 parameters.

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs
Comment (donJoseLuis) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs:51]: can you confirm the error handling model here is to throw to the caller?  If not, would validating parameter state add value?

**Review by donJoseLuis — APPROVED** [review]

**Comment by DavidKarlas** [comment]
Waiting for rebase after IDE Tests PR is merged

**Review by DavidKarlas — APPROVED** [review]

**Review by donJoseLuis — APPROVED** [review]

**Pull request merged**
